### PR TITLE
[bitnami/oauth2-proxy] add useApplicationDefaultCredentials google

### PR DESCRIPTION
**Pull request by franviera92**
### Applicable issues

- fixes #

### Checklist

- [ ] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [ ] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [ ] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [ ] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Comment by carrodher** [comment]
Thanks for your contribution! Could you please bump the chart version in the _Chart.yaml_? This is necessary to test the changes and cut a new release.

**Comment by franviera92** [comment]
> Thanks for your contribution! Could you please bump the chart version in the _Chart.yaml_? This is necessary to test the changes and cut a new release.

Done man

**Comment by franviera92** [comment]
> Thanks for your contribution! Could you please bump the chart version in the _Chart.yaml_? This is necessary to test the changes and cut a new release.

could you helpme to merge? please

File: bitnami/oauth2-proxy/values.yaml [hunk 1]
```diff
@@ -358,6 +358,7 @@ configuration:
   ## @param configuration.google.existingSecret Existing secret containing Google Service Account
   ##
   google:
+    useApplicationDefaultCredentials: false
```
Comment (carrodher) [inline bitnami/oauth2-proxy/values.yaml:361]: You need to add the metadata for this new parameter, see the above metadata corresponding to the already existing parameters.

Comment (franviera92) [reply bitnami/oauth2-proxy/values.yaml:362]: i belive is ok now

**Comment by carrodher** [comment]
Thanks! It seems now the DCO is failing because one of the commits is not signed, you should be able to fix that by doing the following:
1. In your local branch, run: `git rebase HEAD~8 --signoff`
2. Force push your changes to overwrite the branch: `git push --force-with-lease origin main`

**Comment by franviera92** [comment]
> Thanks! It seems now the DCO is failing because one of the commits is not signed, you should be able to fix that by doing the following:
> 
> 1. In your local branch, run: `git rebase HEAD~8 --signoff`
> 2. Force push your changes to overwrite the branch: `git push --force-with-lease origin main`

Thanks DCO is Ok

**Comment by migruiz4** [comment]
Hi @franviera92, 
You accidentally re-signed commits beyond the scope of this PR.

That is why the Diff view is showing changes like the image tag, and the reason why the commit list in the PR shows `Release 2.1.10`.

I'm sorry but I can not approve and merge this PR as it is as this may cause problems in the repository commit history.

I think the easier solution would be to:
- Undo the commits in this PR: `git reset --hard 5a628ec907ceb3b46d2838782928ad5cce74acbb`
- Push your branch `git push --force`
- In the Github UI at 'https://github.com/franviera92/charts', click on the 'Sync fork'.
- Now back to your computer, `git pull`
- Re-apply your changes in the `bitnami/oauth2-proxy` and commit your changes: `git commit -S -s -a -m "[bitnami/oauth2-proxy] add useApplicationDefaultCredentials google"`

To make sure your changes aren't lost in the process I copied them here:
- `values.yaml`
```diff
  ## Alternatively, use an existing secret (see secret-google.yaml for required fields)
  ##
  ## @param configuration.google.enabled Enable Google service account
+ ## @param configuration.google.useApplicationDefaultCredentials Use the application-default credentials (i.e. Workload Identity on GKE) instead of providing a service account JSON
  ## @param configuration.google.adminEmail Google admin email
  ## @param configuration.google.groups Restrict logins to members of these google groups
  ## @param configuration.google.serviceAccountJson Google Service account JSON
  ## @param configuration.google.existingSecret Existing secret containing Google Service Account
  ##
  google:
+   useApplicationDefaultCredentials: false
```
- deployment.yaml
```diff
+            {{- if .Values.configuration.google.useApplicationDefaultCredentials }}
+            - --google-use-application-default-credentials=true
+            {{- else }}
             - --google-service-account-json=/bitnami/oauth2-proxy/conf/google/service-account.json
+            {{- end }}
```
- Do not forget to bump Chart.yaml with a new minor version, since this PR adds a new value:
```diff
-version: 6.0.6
+version: 6.1.0
```

**Comment by franviera92** [comment]
> Hi @franviera92, You accidentally re-signed commits beyond the scope of this PR.
> 
> That is why the Diff view is showing changes like the image tag, and the reason why the commit list in the PR shows `Release 2.1.10`.
> 
> I'm sorry but I can not approve and merge this PR as it is as this may cause problems in the repository commit history.
> 
> I think the easier solution would be to:
> 
> * Undo the commits in this PR: `git reset --hard 5a628ec907ceb3b46d2838782928ad5cce74acbb`
> * Push your branch `git push --force`
> * In the Github UI at 'https://github.com/franviera92/charts', click on the 'Sync fork'.
> * Now back to your computer, `git pull`
> * Re-apply your changes in the `bitnami/oauth2-proxy` and commit your changes: `git commit -S -s -a -m "[bitnami/oauth2-proxy] add useApplicationDefaultCredentials google"`
> 
> To make sure your changes aren't lost in the process I copied them here:
> 
> * `values.yaml`
> 
> ```diff
>   ## Alternatively, use an existing secret (see secret-google.yaml for required fields)
>   ##
>   ## @param configuration.google.enabled Enable Google service account
> + ## @param configuration.google.useApplicationDefaultCredentials Use the application-default credentials (i.e. Workload Identity on GKE) instead of providing a service account JSON
>   ## @param configuration.google.adminEmail Google admin email
>   ## @param configuration.google.groups Restrict logins to members of these google groups
>   ## @param configuration.google.serviceAccountJson Google Service account JSON
>   ## @param configuration.google.existingSecret Existing secret containing Google Service Account
>   ##
>   google:
> +   useApplicationDefaultCredentials: false
> ```
> 
> * deployment.yaml
> 
> ```diff
> +            {{- if .Values.configuration.google.useApplicationDefaultCredentials }}
> +            - --google-use-application-default-credentials=true
> +            {{- else }}
>              - --google-service-account-json=/bitnami/oauth2-proxy/conf/google/service-account.json
> +            {{- end }}
> ```
> 
> * Do not forget to bump Chart.yaml with a new minor version, since this PR adds a new value:
> 
> ```diff
> -version: 6.0.6
> +version: 6.1.0
> ```

I have already made the change, perhaps I need to update the chart.yml, can you confirm?

**Comment by franviera92** [comment]
@fmulero help please :(

File: bitnami/oauth2-proxy/templates/deployment.yaml [hunk 2]
```diff
@@ -99,7 +99,11 @@ spec:
             {{- end }}
             {{- if .Values.configuration.google.enabled }}
             - --google-admin-email={{ .Values.configuration.google.adminEmail }}
+            {{- if .Values.configuration.google.useApplicationDefaultCredentials }}
+            - --google-use-application-default-credentials=true
+            {{- else }}
             - --google-service-account-json=/bitnami/oauth2-proxy/conf/google/service-account.json
++            {{- end }}
```
Comment (fmulero) [inline bitnami/oauth2-proxy/templates/deployment.yaml:106]: There is a little typo here
```suggestion
            {{- end }}
```

**Review by fmulero — CHANGES_REQUESTED** [review]
Thanks a lot @franviera92 for your contribution.

I've just left a little comment due to typo, the rest seems correct. Could you give it a glance?

File: bitnami/oauth2-proxy/templates/deployment.yaml (hunk 2 above)
Comment (franviera92) [reply bitnami/oauth2-proxy/templates/deployment.yaml]: oh sorry, done fix

**Pull request merged**